superset-frontend/src/features/alerts/AlertReportModal.tsx

```diff
@@ -498,9 +493,16 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
   >([]);
   const onNotificationAdd = () => {
     const settings: NotificationSetting[] = notificationSettings.slice();
```

Comment:
I know this is existing code, but since we're reusing this pattern, if we need to make a copy for immutability, I would suggest using destructuring over slice. But if you can use map and filter on notificationSettings, that would be all the better. And those methods also make a copy, so would still work as a pattern for keeping notificationSettings immutable if you needed it.

Reply:
changed the pattern to use filter and reduce instead of slice